lib/evaluation/markForSubmissionEvaluationHandler.ts

```diff
@@ -0,0 +1,114 @@
+        this.iterateOverInputDescriptors(pd, vc, key);
+      }
+    }
+  }
+
+  private iterateOverInputDescriptors(pd: PresentationDefinition, vc: [number, unknown], path: string): void {
+    const error = this.results.find(
+      (result) => result.status === Status.ERROR && result.verifiable_credential_path === `$.${path}[${vc[0]}]`
+    );
+    if (error) {
+      this.results.push({
+        ...error,
+        message: 'The input candidate is not eligible for submission',
+      });
+    } else {
+      this.createPresentationSubmission(pd, vc, path);
+    }
+  }
+
+  private createPresentationSubmission(pd: PresentationDefinition, vc: any, path: string) {
+    this.verifiablePresentation.presentationSubmission.definition_id = pd.id;
+    const info = this.results.filter((result) => result.verifiable_credential_path === `$.${path}[${vc[0]}]`);
+    if (!this.verifiablePresentation[`${path}`]) {
+      this.verifiablePresentation[`${path}`] = [];
+    }
+    this.handleInputDescriptors(pd.input_descriptors, vc, info, path);
+  }
+
+  private handleInputDescriptors(
+    inputDescriptors: InputDescriptor[],
+    vc: [number, any],
+    info: HandlerCheckResult[],
+    path: string
+  ) {
+    for (const id of inputDescriptors.entries()) {
+      for (const r of info) {
+        if (r.input_descriptor_path === `$.input_descriptors[${id[0]}]`) {
+          const descriptor: Descriptor = { id: id[1].id, format: 'ldp_vc', path: r.verifiable_credential_path };
+          this.pushToDescriptorsMap(descriptor, vc, path);
+          this.pushToResults(r, id);
+        }
+      }
+    }
+  }
+
+  private pushToResults(r: HandlerCheckResult, id: [number, InputDescriptor]) {
+    this.results.push({
+      input_descriptor_path: r.input_descriptor_path,
+      verifiable_credential_path: r.verifiable_credential_path,
+      evaluator: this.getName(),
+      status: Status.INFO,
+      payload: { group: id[1].group },
+      message: 'The input candidate is eligible for submission',
+    });
+  }
+
+  private pushToDescriptorsMap(newDescriptor: Descriptor, vc: [number, any], path: string) {
+    const descriptorMap: Descriptor[] = this.verifiablePresentation.presentationSubmission.descriptor_map;
+    if (descriptorMap.find((d) => d.id === newDescriptor.id && d.path !== newDescriptor.path)) {
+      this.verifiablePresentation[`${path}`].push(vc[1]);
+      this.verifiablePresentation.presentationSubmission.descriptor_map.forEach((d: Descriptor) =>
+        this.addPathNested(d, newDescriptor)
+      );
+    } else if (
+      !descriptorMap.find(
+        (d) => d.id === newDescriptor.id && d.format === newDescriptor.format && d.path === newDescriptor.path
+      )
+    ) {
+      this.verifiablePresentation[`${path}`].push(vc[1]);
+      this.verifiablePresentation.presentationSubmission.descriptor_map.push(newDescriptor);
+    }
+  }
+
+  private addPathNested(descriptor: Descriptor, nestedDescriptor: Descriptor): Descriptor {
```

Comment:
rename method to `addPathNestedDescriptor` perhaps ?

Reply:
renamed